data-service/src/main/java/com/amazonaws/blox/dataservice/storage/DeploymentDDBStore.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.dataservice.storage;
+
+import com.amazonaws.blox.dataservice.model.Deployment;
+import com.amazonaws.blox.dataservice.model.DeploymentStatus;
+import com.amazonaws.blox.dataservice.storage.model.DeploymentDDBRecord;
+import com.amazonaws.blox.dataservice.storage.model.DeploymentDDBRecordTranslator;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBScanExpression;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public class DeploymentDDBStore implements DeploymentStore {
+
+  @NonNull private DynamoDBMapper dynamoDBMapper;
+
+  @Override
+  public Deployment createDeployment(final Deployment deployment) {
+    final DeploymentDDBRecord record =
+        DeploymentDDBRecordTranslator.toDeploymentDDBRecord(deployment);
+    dynamoDBMapper.save(record);
+    return DeploymentDDBRecordTranslator.fromDeploymentDDBRecord(record);
+  }
+
+  @Override
+  public Deployment updateDeployment(final Deployment deployment) {
+    DeploymentDDBRecord record =
+        dynamoDBMapper.load(DeploymentDDBRecord.withHashKey(deployment.getDeploymentId()));
```

Comment:
It has a higher chance of success if we retrieve the latest version before updating it.